crates/net/eth-wire/src/ethstream.rs

```diff
@@ -15,45 +16,45 @@ use std::{
     pin::Pin,
     task::{Context, Poll},
 };
-use tokio_stream::Stream;
+use tokio::sync::{mpsc, oneshot};
+use tokio_stream::{wrappers::UnboundedReceiverStream, Stream};
 
 /// [`MAX_MESSAGE_SIZE`] is the maximum cap on the size of a protocol message.
 // https://github.com/ethereum/go-ethereum/blob/30602163d5d8321fbc68afdcbbaf2362b2641bde/eth/protocols/eth/protocol.go#L50
 pub const MAX_MESSAGE_SIZE: usize = 10 * 1024 * 1024;
 
 /// An un-authenticated [`EthStream`]. This is consumed and returns a [`EthStream`] after the
 /// `Status` handshake is completed.
-#[pin_project]
-pub struct UnauthedEthStream<S> {
-    #[pin]
-    inner: S,
+pub struct UnauthedEthStream {
+    p2p_tx: mpsc::Sender<Bytes>,
+    p2p_rx: mpsc::UnboundedReceiver<Bytes>,
+    p2p_disconnect: oneshot::Sender<DisconnectReason>,
+    req_handler_tx: mpsc::UnboundedSender<CapsMessage>,
+    req_handler_rx: mpsc::UnboundedReceiver<CapsMessage>,
```

Comment:
it is not clear to me what we use these for

Reply:
these are returned to store in an active session. an eth message travels from active session via `to_session` -> `active_session_rx` in session manager which is polled in `Swarm`, which in turn is polled by `NetworkManager` which sends eth message to the `EthRequestHandler` via the `to_eth_request_handler` channel. So, transitively, the `req_handler_tx` goes to the `EthRequestHandler`.

Reply:
at second look, `network_manager_tx` and `network_manager_rx` are more suitable names as that is where [messages are sent to different handlers based on their `PeerMessage` variant](https://github.com/paradigmxyz/reth/blob/171166e84ae75f2129d9c3a9abe504d574d2c587/crates/net/network/src/manager.rs#L469-L507) and only type `EthRequest` goes to `EthRequestHandler`, my bad

Reply:
more clearly, `network_manager_tx_closure` or `network_manager_tx_transitive` to denote the transitive edge between a capability stream and the network manager